portfolio/src/main/java/com/google/sps/data/RequestJson.java

```diff
@@ -0,0 +1,240 @@
+//  RequestJson.java
+//  Sample Google API
+//
+//  Created by Mohammadreza on 10/7/18.
+//  Copyright © 2019 Mohammadreza Mohades. All rights reserved.
+
+package com.google.sps.data;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.data.BookFieldsEnum;
+import com.google.sps.data.FullBook;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.IllegalAccessException;
+import java.lang.NoSuchFieldException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class RequestJson {
+  public RequestJson() {}
+
+  public JSONObject requestQuery(String query) throws Exception {
+    // This method calls the actual Book API from the URL and returns a JSON Object from the
+    // response convert space to url format
+
+    String url = String.format("https://www.googleapis.com/books/v1/volumes?q=%s", query);
+    // this link contains different data than the book's selfLink
+    URL obj = new URL(url);
+    HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+    con.setRequestMethod("GET");
+    con.setRequestProperty("User-Agent", "Mozilla/5.0");
+    int responseCode = con.getResponseCode();
+
+    BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+    String inputLine;
+    StringBuffer response = new StringBuffer(); // will contain the large string of information
+    while ((inputLine = in.readLine()) != null) {
+      response.append(inputLine);
+    }
+    in.close();
+
+    JSONObject myResponse = new JSONObject(response.toString());
+    return myResponse;
+  }
+
+  public ArrayList<FullBook> call_me(String query, int numberResults) throws Exception {
+    File index = new File("output");
+
+    if (!index.exists()) {
+      index.mkdir();
+    } else {
+      String[] entries = index.list();
+      for (String s : entries) {
+        File currentFile = new File(index.getPath(), s);
+        currentFile.delete();
+      }
+    }
+
+    ArrayList<FullBook> returnedList = new ArrayList<FullBook>();
+
+    JSONObject myResponse =
+        new JSONObject(requestQuery(query).toString()); // request Query returns a JSON object of
+                                                        // the JSON string with all the data
+    JSONArray allItems = myResponse.getJSONArray("items");
+
+    for (int i = 0; i < numberResults; i++) {
+      returnedList.add(jsonToFullBook(allItems.getJSONObject(i))); // jsonToFullBook
+    }
+    return returnedList;
+  }
+
+  public FullBook jsonToFullBook(JSONObject jsonObject) throws JSONException {
```

Comment:
optional: this is a really long function – you can consider writing smaller private helper functions for readability for future functions if they're long :)